test/e2e/framework/alertmanager.go

```diff
@@ -174,22 +174,46 @@ func amImage(version string) string {
 
 func (f *Framework) WaitForAlertmanagerInitializedMesh(name string, amountPeers int) error {
 	return f.Poll(time.Second*20, time.Second, func() (bool, error) {
-		request := f.ProxyGetPod(name, "9093", "/api/v1/status")
-		resp, err := request.DoRaw()
+		amStatus, err := f.GetAlertmanagerConfig(name)
 		if err != nil {
 			return false, err
 		}
+		if len(amStatus.Data.MeshStatus.Peers) == amountPeers {
+			return true, nil
+		}
+
+		return false, nil
+	})
+}
+
+func (f *Framework) GetAlertmanagerConfig(n string) (alertmanagerStatus, error) {
+	var amStatus alertmanagerStatus
+	request := f.ProxyGetPod(n, "9093", "/api/v1/status")
+	resp, err := request.DoRaw()
+	if err != nil {
+		return amStatus, err
+	}
+
+	if err := json.Unmarshal(resp, &amStatus); err != nil {
+		return amStatus, err
+	}
+
+	return amStatus, nil
+}
 
-		var amStatus alertmanagerStatus
-		if err := json.Unmarshal(resp, &amStatus); err != nil {
+func (f *Framework) WaitForAlertmanagerResolveTimeoutConfig(amName string, resolveTimeout int) error {
+	return f.Poll(time.Minute*5, time.Second, func() (bool, error) {
+		config, err := f.GetAlertmanagerConfig("alertmanager-" + amName + "-0")
+		if err != nil {
 			return false, err
 		}
 
-		if len(amStatus.Data.MeshStatus.Peers) == amountPeers {
-			return true, nil
+		if config.Data.ConfigJSON.Global.ResolveTimeout != 6*60*1000000000 {
```

Comment:
It's very hard to understand what's happening here. It would be better if we just compare the actually loaded [config](https://github.com/prometheus/alertmanager/blob/fced9e126b334fe0b9eb6733c4138638459a4bda/api/api.go#L171). We can just compare the loaded config string with the expected one, instead of some arbitrary value that's a subset of the config.